### DISCUSSION TITLE
Feature Request: Nades per team config

### DISCUSSION THREAD

**Discussion by trollollol123** [yonilerner/cs2-retakes-allocator#21] [Ideas]
Give the server owners a bit more control of how the rounds are played.

Let us decide how many of each nade type we want to give to each team.
This should also be map based for maximum usefulness. Maybe on some maps, T having 2 smokes for example is too much.
It also gives a way to mix things up a bit and make each server more unique.
same with awp (where maybe VIP has prio)

I let you figure out how exactly to make the configs.  
I think in CSGO we had CFGs with mapnames and then commands inside each. And then correct cfg was loaded when mapchanged.

de_dust2.cfg
de_mirage.cg

de_dust2.cfg
sm_retakes_weapon_nades_smokegrenade_ct_max 0
sm_retakes_weapon_nades_smokegrenade_t_max 0
sm_retakes_weapon_nades_flashbang_ct_max 2
sm_retakes_weapon_nades_flashbang_t_max 1
sm_retakes_weapon_nades_molotov_ct_max 1
sm_retakes_weapon_nades_molotov_t_max 1
sm_retakes_weapon_nades_hegrenade_ct_max 1
sm_retakes_weapon_nades_hegrenade_t_max 1
sm_retakes_weapon_awp_t_max 1
sm_retakes_weapon_awp_ct_max 1

de_mirage.cg
sm_retakes_weapon_nades_smokegrenade_ct_max 2
sm_retakes_weapon_nades_smokegrenade_t_max 0
sm_retakes_weapon_nades_flashbang_ct_max 2
sm_retakes_weapon_nades_flashbang_t_max 3
sm_retakes_weapon_nades_molotov_ct_max 1
sm_retakes_weapon_nades_molotov_t_max 1
sm_retakes_weapon_nades_hegrenade_ct_max 0
sm_retakes_weapon_nades_hegrenade_t_max 1
sm_retakes_weapon_awp_t_max 1
sm_retakes_weapon_awp_ct_max 1

**Comment by yonilerner** [comment]
Seems reasonable

**Reply by yonilerner** [reply]
Ill probably also add global defaults

**Comment by yonilerner** [comment]
@BrutalCSEjziponken Do you want to try out this build and see if it works well for you? https://github.com/yonilerner/cs2-retakes-allocator/pull/98

**Comment by Micka2302** [comment]
Hi, 

Any news from this feature? 
At the moment, the allocator plugins give too many nades.

**Reply by yonilerner** [reply]
The PR linked above will be ready to go very soon. Sorry for the delays

**Reply by Micka2302** [reply]
No problem, everyone has their priorities :)

**Comment by yonilerner** [comment]
https://github.com/yonilerner/cs2-retakes-allocator/releases/tag/v1.3.0 has the first version of nade configs. Let me know what you think

**Reply by Micka2302** [reply]
Hi, I’ m going to try this.

**Reply by Micka2302** [reply]
Feedback:

I've noted 1 point about grenade update. 

At times, a T or CT has 2 grenades in the same round. 

Whereas in the setup I have 1 grenade, so it's 1 grenade of each type allowed and so the player can have several grenades?
If this is the case, I think we'd need a type variable to limit the number of grenades that a single player can have.

```
  "MaxNades": {
    "GLOBAL": {
      "Terrorist": {
        "Flashbang": 1,
        "Smoke": 1,
        "Molotov": 1,
        "HighExplosive": 1
      },
      "CounterTerrorist": {
        "Flashbang": 1,
        "Smoke": 1,
        "Molotov": 1,
        "HighExplosive": 1
      }
    }
  },

```
![20240311181716_1](https://github.com/yonilerner/cs2-retakes-allocator/assets/20491694/5df0f207-16a4-4cbc-a46a-4487ac4dee25)

I've also made another suggestion: limit awp to a minimum number of players (for example, awp is allowed from 4 or 5 players, depending on the configuration).

**Reply by yonilerner** [reply]
Ya, I can add a separate config for max nades per player

**Reply by yonilerner** [reply]
The next release will have a config called `MaxTeamNades` that should help with this. Let me know what you think

**Comment by yonilerner** [comment]
Closing this since I think the feature has been resolved
